File: aquasense_backend/models/product.py
"""
Product Model - Marketplace items
"""
from sqlalchemy import Column, String, Float, Integer, DateTime, Text
from datetime import datetime
from config.database import Base
import uuid


class Product(Base):
    __tablename__ = "products"

    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    name = Column(String(255), nullable=False)
    category = Column(String(100), nullable=False)  # feed, medicine, equipment, etc.
    description = Column(Text, nullable=True)
    price = Column(Float, nullable=False)
    unit = Column(String(50), nullable=False)  # kg, liters, pieces, etc.
    stock_quantity = Column(Integer, default=0)
    image_url = Column(String(500), nullable=True)
    manufacturer = Column(String(255), nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

File: app/src/main/java/com/parth/aquasense/domain/model/Product.kt
package com.parth.aquasense.domain.model

/**
 * Domain model representing a product in the marketplace.
 *
 * Products can be of different categories (feed, medicine, equipment)
 * and are available for purchase by farmers.
 */
data class Product(
    val id: String,
    val name: String,
    val category: String,
    val description: String?,
    val price: Double,
    val unit: String,
    val stockQuantity: Int,
    val imageUrl: String?,
    val manufacturer: String?,
    val createdAt: String,
    val updatedAt: String
)
